### ISSUE TITLE
lint: add builtin shadowing detection

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#87]
The `builtin-shadow` checker should detect assignments to builtin function names.

For example, this code triggers two warnings:
```go
close := 1
len := close + 2
```
